pkg/adsc/adsc.go

```diff
@@ -952,3 +1074,54 @@ func (a *ADSC) GetEndpoints() map[string]*endpoint.ClusterLoadAssignment {
 	defer a.mutex.Unlock()
 	return a.eds
 }
+
+func (a *ADSC) handleMCP(gvk []string, rsc *any.Any, valBytes []byte) error {
+	if len(gvk) != 3 {
+		return nil // Not MCP
+	}
+	// Generic - fill up the store
+	if a.Store != nil {
+		m := &mcp.Resource{}
+		err := types.UnmarshalAny(&types.Any{
+			TypeUrl: rsc.TypeUrl,
+			Value:   rsc.Value,
+		}, m)
+		if err != nil {
+			return err
+		}
+		val, err := mcpToPilot(m)
+		if err != nil {
+			return err
+		}
+		val.GroupVersionKind = resource.GroupVersionKind{gvk[0], gvk[1], gvk[2]}
+		if err != nil {
+			adscLog.Warna("Invalid data ", err, " ", string(valBytes))
+		} else {
+			cfg := a.Store.Get(val.GroupVersionKind, val.Name, val.Namespace)
+			if cfg == nil {
+				_, err = a.Store.Create(*val)
+				if err != nil {
+					return err
+				}
+			} else {
+				_, err = a.Store.Update(*val)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		if a.LocalCacheDir != "" {
+			strResponse, err := json.MarshalIndent(val, "  ", "  ")
```

Comment:
need to use jsonpb

Reply:
All the logging in this file uses json - yes, will probably be better to use jsonpb. 
BTW - the response is an MCP object, which I think is a gogo proto - so once again mixing libraries...